### PR TITLE
Regen for abstract base class

### PR DESCRIPTION
**Pull request by Yao725**
Part of https://github.com/Azure/autorest.csharp/pull/2520

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Generated/Models/KnowledgeStoreProjectionSelector.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@
 namespace Azure.Search.Documents.Indexes.Models
 {
     /// <summary> Abstract class to share properties between concrete selectors. </summary>
-    public partial class KnowledgeStoreProjectionSelector
```
Comment (Yao725) [inline sdk/search/Azure.Search.Documents/src/Generated/Models/KnowledgeStoreProjectionSelector.cs:14]: This kind of change in `Azure.Search.Documents` comes from that I unified the judgement condition of abstract class in [ModelWriter](https://github.com/Azure/autorest.csharp/blob/0eec8c79d213dbabe38de654f6c9566113b3425b/src/AutoRest.CSharp/Common/Generation/Writers/ModelWriter.cs#L300) and [SerializationWriter](https://github.com/Azure/autorest.csharp/blob/0eec8c79d213dbabe38de654f6c9566113b3425b/src/AutoRest.CSharp/Common/Generation/Writers/SerializationWriter.cs#L188) of autorest.csharp.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/Models/DataSourceCredentialEntity.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 namespace Azure.AI.MetricsAdvisor.Administration
 {
     /// <summary> The DataSourceCredential. </summary>
-    public partial class DataSourceCredentialEntity
+    public abstract partial class DataSourceCredentialEntity
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/Models/DataSourceCredentialEntity.cs:14]: While I agree this should have been made `abstract` during development, this class has already been shipped as part of the Metrics Advisor public API, so we can't change it without architect review. Could we somehow skip these changes for Metrics Advisor?

The same reasoning is true for the `MetricFeedback` and the `NotificationHook` classes.

**Review by kinelski — CHANGES_REQUESTED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/Models/DataSourceCredentialEntity.cs (hunk 2 above)
Comment (Yao725) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/Models/DataSourceCredentialEntity.cs]: Reverted this changes for Metrics Advisor.

**Review by kinelski — DISMISSED** [review]
Metrics Advisor LGTM.

**Pull request merged**